CONTRIBUTING.md

```diff
@@ -194,10 +201,57 @@ $ git push origin my-branch
 Go to https://github.com/yourusername/node and select your branch.
 Click the 'Pull Request' button and fill out the form.
 
-Pull requests are usually reviewed within a few days. If there are comments
-to address, apply your changes in a separate commit and push that to your
-branch. Post a comment in the pull request afterwards; GitHub does
-not send out notifications when you add commits.
+Pull requests are usually reviewed within a few days.
+
+### Step 7: Discuss and update
+
+You will probably get feedback or requests for changes to your Pull Request,
+this is a big part of the submission process, so don't be disheartened!
+
+When someone suggests changes, you can address them by applying them to your
+branch. When your branch changes, Github will automatically update the PR.
+
+You can push more commits to your branch:
+
+```text
+$ git add my/changed/files
+$ git commit
+$ git push origin my-branch
+```
+
+Or you can rebase against master:
+
+```text
+$ git fetch --all
+$ git rebase origin/master
+$ git push --force origin my-branch
+```
+
+Or you can amend the last commit (for example if you want to change the commit
+log).
+
+```text
+$ git add any/changed/files
+$ git commit --amend
+$ git push --force origin my-branch
+```
+
+**Important:** The `git push --force` command is one of the few ways to delete
+history in git, before you use it, make sure you understand the risks. If in
+doubt, you can always ask for guidance in the Pull Request or on IRC.
+
+If your Pull Request has gone quiet, post a comment in the pull request to ping
```

Comment:
Nit: I don't much care whether we go with `Pull Request` or `pull request`, but please use one or the other exclusively. Both are used in this sentence. :-D 

Nit: `has gone quiet` may be a little colloquial. Maybe `If your Pull Request needs attention, post a comment...`?